lib/pbench/server/api/resources/query_controllers.py

```diff
@@ -0,0 +1,173 @@
+from flask import request
+from flask_restful import Resource, abort
+from copy import deepcopy
+import requests
+
+from dateutil import parser, relativedelta
+
+
+class QueryControllers(Resource):
+    """
+    Abstracted Pbench API to get date-bounded controller data.
+    """
+
+    def __init__(self, app, api):
+        self.app = app
+        self.api = api
+
+    @staticmethod
+    def _genMonthUriRange(prefix, index, start, end):
+        """
+        _genMonthUriRange Construct a comma-separated list of index names
+        qualified by year and month suitable for use in the Elasticsearch
+        /_search query URI.
+
+        The month is incremented by 1 from "start" to "end"; for example,
+
+        _genMonthUriRange('drb.', 'v5.run.', '2020-08', '2020-10') will result
+        in
+
+            'drb.v4.run.2020-08,drb.v4.run.2020-09,drb.v4.run.2020-10,'
+
+        This code is adapted from Javascript in the Pbench dashboard's
+        'moment_constants.js' getAllMonthsWithinRange function.
+
+        Args:
+            prefix ([type]): The Pbench index prefix assigned to the dashboard
+            index ([type]): The desired monthly index pattern
+            start ([type]): The start time (generally 'YYYY-MM')
+            end ([type]): The end time (generally 'YYYY-MM')
+
+        Returns:
+            [string]: A comma-separated list of month-qualified index names
+        """
+        startDate = parser.parse(start)
+        endDate = parser.parse(end)
+        referenceDate = deepcopy(startDate)
+        monthResults = list()
+        queryString = ""
+        oneMonth = relativedelta.relativedelta(months=1)
+        if startDate and endDate:
+            while referenceDate < endDate:
```

Comment:
Can we use a for loop here so we dont need to keep the copy of referenceDate there is a [rrule](http://labix.org/python-dateutil#head-470fa22b2db72000d7abe698a5783a46b0731b57) module in `python-dateutil` that we can use to get the range of months for example 
```
for dt in rrule.rrule(rrule.MONTHLY, dtstart=start, until=end):
    print(dt)
```

Reply:
Cool. I did not know that.